src/http/session.rs

```diff
@@ -445,14 +477,29 @@ impl Stream for DefaultStream {
     }
 
     fn set_headers<'n, 'v>(&mut self, headers: Vec<Header<'n, 'v>>) {
```

Comment:
I think this could be done a bit more simply and efficiently.

Just doing a match on `self.headers` and then:

- `None` => overwrite with `Self(new_headers.collect())`
- `Some(ref mut old_headers) => old_headers.extend(new_headers)`

(`new_headers` would be the mapped-over iterator)

Do you think you could do this tweak? It's not a major blocker for merging this, but would bit a bit nicer...